### PR TITLE
Utilizing Tests for React Calculator Site

### PR DESCRIPTION
**Pull request by JohnHernCode**
Add tests using Jest and Snapshots

### CODE REVIEW

**Review by trekab — COMMENTED** [review]
### Invalid! :stop_sign: 
Hi @JohnHernCode! Unfortunately, I have to invalidate your PR because you didn't follow the correct [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) as expected on this project.

_Please make the necessary changes and submit another code review request._
__[Treasure Kabareebe](https://trekab.github.io)__

**Comment by JohnHernCode** [comment]
For some reason, I made the mistake of creating the pull request from master using Git Kraken, I made the correct changes. Git flow was used when creating the feature/testing branch.

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello @JohnHernCode,

Good job with your tests so far, I will mention you some things that need to be changed in order to get this project approved: 

- [x] All functionalities are tested using [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/), `@testing-library/react`, see the [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/testing/useReactTestingLibrary.png) from [Pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106580952/). (Currently you are using enzyme and react-test-renderer in order to test your project, please get rid of them and use only `@testing-library/react`
- [x] All functions in calculate.js file are covered with unit tests (You are missing to test some of your calculator functionalities like +/- and error handling like dividing by 0.
- [x] Dont forget about negative tests.

Please correct those issues and send another review request, good luck, and well done so far!

**Comment by JohnHernCode** [comment]
- All functionalities are tested using React Testing Library, @testing-library/react, see the screenshot from Pathwright. (Currently you are using enzyme and react-test-renderer in order to test your project, please get rid of them and use only @testing-library/react

**_Removed enzyme and snapshots and used only @testing-library/react_**

- All functions in calculate.js file are covered with unit tests (You are missing to test some of your calculator functionalities like +/- and error handling like dividing by 0.

**_Added tests for +/- and divide by 0_**

- Dont forget about negative tests.

**_Added negative tests_**

File: package.json [hunk 1]
```diff
@@ -8,12 +8,15 @@
     "@testing-library/user-event": "^13.1.9",
     "big-js": "^3.1.3",
     "big.js": "^6.1.1",
```
Comment (addod19) [inline package.json]: [Optional]
- Kindly remove one of the ` big.js` packages. :+1:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

HI @JohnHernCode, congratulations on this project :smile: The negative test case added :+1: All functionalities are tested using RTL :100: However, one of the requirements was not met. Let us address the comment below:

- The division by zero logic was not addressed. For example when I enter `8 / 0` the expected answer should be `cannot divide by 0` but in our case, we get the result as `0`, Which mathematically is incorrect. Let us address it and write a test for it
- Also since this project focuses on testing, we are encouraged to add instructions about how to run the test in our README file :smile: 

Once these are addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: package.json (hunk 1 above)
Comment (JohnHernCode) [reply package.json]: Removed.

**Comment by JohnHernCode** [comment]
Fixed the test for divide by zero and the operation itself. The result is now "cannot divide by 0" :) 

Added the test instructions to the readme file. 

Removed "big-js": "^3.1.3", from the packages.json file.

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 
@JohnHernCode Welldone for implementing the suggested changes from the Previous Code Reviewer. I Must commend your effort. however, when I run your code I found out that your `calculate.test.js`  is failing due to the fact that you have removed `big.js`.here the image 
[error](https://user-images.githubusercontent.com/38242590/125669071-7a08bea3-db6d-4c8a-a283-1fbd1a7cfdd4.png).
kindly fix and ensure all your test passes

### once fixed kindly request for another review

**Comment by JohnHernCode** [comment]
It looks like I removed the wrong one. I fixed the package.json, retested, and everything passed.

**Review by rindrajosia — APPROVED** [review]
STATUS: APPROVED :heavy_check_mark: :heavy_check_mark:

Good job :tada: . This project is approved. kindly merge and prepare for the next task. Thanks

Nice to have [OPTIONAL] :

- [ ] Please update your live demo because in your live demo when I enter 8 / 0 the expected answer should be `cannot divide by 0` but in our case, we get the result as 0, Which mathematically is incorrect. In addition to that, make sure that the screenshot is not broken as your readme is the first thing users check.

**Pull request merged**